app/cdap/components/RecipeList/index.tsx

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useEffect, useReducer } from 'react';
+import { RecipesTable } from './RecipesTable';
+import PaginationStepper from 'components/shared/PaginationStepper';
+import { RecipeTableDiv, PaginationContainer } from './styles';
+import { IRecipe, SortBy, SortOrder } from './types';
+import { reducer, nextPage, prevPage, getSavedRecipes, defaultInitialState } from './reducer';
+
+interface IRecipeListProps {
+  isOpen: boolean;
+  showAllColumns?: boolean;
+  showActions?: boolean;
+  showPagination?: boolean;
+  enableSorting?: boolean;
+  sortBy?: SortBy;
+  sortOrder?: SortOrder;
+  pageSize?: number;
+  viewHandler?: (selectedRecipe: IRecipe) => void;
+  editHandler?: (selectedRecipe: IRecipe) => void;
+  selectHandler?: (selectedRecipe: IRecipe) => void;
+}
+
+const RecipeList = ({
+  isOpen,
+  showAllColumns,
+  showActions,
+  showPagination,
+  enableSorting,
+  sortBy,
+  sortOrder,
+  pageSize,
+  viewHandler,
+  editHandler,
+  selectHandler,
+}: IRecipeListProps) => {
+  const [state, dispatch] = useReducer(reducer, {
+    ...defaultInitialState,
+    pageLimit: pageSize,
+    sortColumn: sortBy,
+    sortOrder,
+  });
+
+  const { recipes } = state;
+
+  const Pagination = ({}) => {
+    return (
+      <PaginationContainer className="float-left">
+        <div>
+          <PaginationStepper
+            onNext={() => nextPage(dispatch, state)}
+            onPrev={() => prevPage(dispatch, state)}
+            nextDisabled={!state.nextPageToken}
+            prevDisabled={!state.previousTokens.length}
+          />
+        </div>
+      </PaginationContainer>
+    );
+  };
+
+  const viewRecipeHandler = (selectedRecipeObject: IRecipe) => {
+    if (viewHandler) {
```

Comment:
Will there be scenarios where these handlers are not defined/passed?

Reply:
Yes, as this component is used at different places, when it is shown in Wrangler new home page, there won't be any handlers. So added the checks here.